### PR TITLE
Expanding on how routing topologies work

### PR DESCRIPTION
**Pull request by WojcikMike**
Fixes https://github.com/Particular/NServiceBus.RabbitMQ/issues/183

@Particular/docs-maintainers @Particular/rabbitmq-transport-maintainers please have a look if this approach make sense? this PR still need more work but I want to show which direction I am heading to minimize the chances that I am doing something in a sub-optimal manner

### CODE REVIEW

**Comment by bording** [comment]
While the examples seem like they could be useful, I think we need to also add a bit more around describing the actual topology set up first before diving into an example.

Also, all of this might end up benefiting from diagrams/images vs just trying to describe in text.

**Comment by WojcikMike** [comment]
@bording I changed my description to match what we talked about on monday. Please have a look

File: nservicebus/rabbitmq/routing-topology.md [hunk 1]
```diff
@@ -10,15 +10,32 @@ The RabbitMQ transport has the concept of a routing topology, which controls how
 
 ## Conventional Routing Topology
 
-By default, the RabbitMQ transport uses the `ConventionalRoutingTopology`, which creates separate [fanout exchanges](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchange-fanout) for each message type, including inherited types, being published in the system. This means that polymorphic routing and multiple inheritance for events is supported since each subscriber will bind its input queue to the relevant exchanges based on the event types that it has handlers for.
+By default, the RabbitMQ transport uses the `ConventionalRoutingTopology`, which creates separate [fanout exchanges](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchange-fanout) for each message type being published in the system and for each endpoint. This means that polymorphic routing and multiple inheritance for events is supported since each subscriber will bind its input queue to the relevant exchanges based on the event types that it has handlers for.
```
Comment (bording) [inline nservicebus/rabbitmq/routing-topology.md]: The "This means that polymorphic routing and multiple inheritance.." sentence doesn't really belong here any more, now that we'll have a separate section on publishing.

Also, we should be able to handle this with the direct topology as well (caveat on multiple interfaces aside) so I'm not really sure what this is trying to point out anyway.

**Comment by bording** [comment]
@WojcikMike This looks like a good step towards what we discussed. :+1:

**Comment by weralabaj** [comment]
@WojcikMike LGTM. Are there any changes across versions? Seems like all it's written now applies to versions 2.0+ inclusive.

**Comment by WojcikMike** [comment]
@weralabaj from what I know our routing strategies stayed the same in terms of interacting with broker

File: nservicebus/rabbitmq/routing-topology.md (hunk 1 above)
Comment (WojcikMike) [inline nservicebus/rabbitmq/routing-topology.md]: At the moment We are not supporting polymorphic routing in direct topology so it seems fair to state it in conventional. Basically there is a different behavior between those 2 and I would like to capture it. 

For now I moved that sentence next to publish.

**Comment by weralabaj** [comment]
@WojcikMike is that actually ready to go? seems pretty good already :)

**Comment by WojcikMike** [comment]
@weralabaj Let's give 1 day for @bording and @adamralph to review. Tomorrow morning we can merge it if no more feedback.

**Comment by WojcikMike** [comment]
@adamralph take a look at this: https://github.com/Particular/docs.particular.net/pull/1952/commits/912df66f64f23407be2995eabc0953729f30ed94

Covers the max length thing

**Comment by adamralph** [comment]
Thanks @WojcikMike. I reworded the sentence, please see https://github.com/Particular/docs.particular.net/pull/1952/commits/ddd24f5bf9cc11fd116e7909fd34e0874ab47cb9

I'm not if the style of calling out the warning ("WARNING" in caps) fits the style guidelines for docs or whether it should be, e.g. in lower case bold, with a colon, etc., but I'll let the @Particular/docs-maintainers comment on that.

**Comment by WojcikMike** [comment]
Dear [elders of doco](https://github.com/orgs/Particular/teams/docs-maintainers) can you please have a look, and merge if you are happy

cc: @Particular/docs-maintainers

File: nservicebus/rabbitmq/routing-topology.md [hunk 2]
```diff
@@ -8,16 +8,43 @@ versions: '[2,]'
...
 
 ## Direct Routing Topology
 
-The `DirectRoutingTopology` is another provided routing topology that routes all events through a single exchange, `amq.topic` by default. Events are published using a routing key based on the event type, and subscribers will use that key to filter their subscriptions.
+The `DirectRoutingTopology` routes all events through a single exchange, `amq.topic` by default. Events are published using a routing key based on the event type, and subscribers will use that key to filter their subscriptions.
+
+
+### Sending using Direct Routing Topology
+
+Every endpoint creates a queue with name that is equal to the endpoint name. When an endpoint sends a message it sends it to a default exchange with a routing key equal to the destination endpoint name. This makes use of RabbitMQ [default exchanges](https://www.rabbitmq.com/tutorials/amqp-concepts.html) to move the message to a queue with the same name.
```
Comment (SimonCropp) [inline nservicebus/rabbitmq/routing-topology.md]: perhaps "creates a queue with a name "?

File: nservicebus/rabbitmq/routing-topology.md [hunk 3]
```diff
@@ -8,16 +8,43 @@ versions: '[2,]'
...
+
+
+### Publishing using Direct Routing Topology
+
+Every endpoint publishes an event using the `amq.topic` exchange with a routing key of the form 'Namespace.TypeName', corresponding to the type of the event. The event is moved to all queues that have a binding for that event type.
+
+An endpoint that subscribes to a given event creates a binding to the default exchange with the appropriate routing key.
+
+WARNING: In accordance with the [AMQP 0.9.1 standard](http://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.publish.routing-key) the routing key has a length limit of 255 characters.
```
Comment (SimonCropp) [inline nservicebus/rabbitmq/routing-topology.md]: for consistency lets make this url https

**Comment by SimonCropp** [comment]
very minor comments. apart from that LGTM

**Comment by WojcikMike** [comment]
@SimonCropp changes incorporated

**Pull request merged**